app/src/main/java/it/quip/android/model/BaseParseObject.java

```diff
@@ -0,0 +1,25 @@
+package it.quip.android.model;
+
+import com.parse.ParseObject;
+
+import it.quip.android.util.MockUtils;
+
+
+public class BaseParseObject extends ParseObject {
+
+    protected void safePut(String key, Object value) {
+        if (value != null) {
+            this.put(key, value);
+        }
+    }
+
+    @Override
+    public String getObjectId() {
+        String objectId = super.getObjectId();
+        if (null == objectId) {
+            objectId = MockUtils.randomId();
```

Comment:
This is temporary. Its there only to make mock objects work. Since we use getObjectId everywhere, and Parse object ids only get set on save (which mock objects don't do), any call to getObjectId usually returns null. To mitigate that, I put in this case statement. We can do two things here:
1. Wrap it in a mock flag and keep it.
2. Get rid of it and figure out how to attach object IDs to mock objects in offline mode (which Parse complains about).